odoo/orm/models.py

```diff
@@ -3571,8 +3567,16 @@ def _fetch_query(self, query: Query, fields: Sequence[Field]) -> Self:
                     sql = self._field_to_sql(self._table, field.name, query)
                     sql = SQL("pg_size_pretty(length(%s)::bigint)", sql)
                 else:
+                    sql = self._field_to_sql(self._table, field.name, query)
                     # flushing is necessary to retrieve the en_US value of fields without a translation
-                    sql = self._field_to_sql(self._table, field.name, query, flush=bool(field.translate))
+                    # otherwise, re-create the SQL without flushing
+                    if not field.translate:
+                        to_flush = list(sql.to_flush)
+                        with contextlib.suppress(ValueError):
+                            # ValueError raised when we cannot remove the current field
+                            to_flush.remove(field)
+                            sql = SQL(sql.code, *sql.params)
+                            sql._SQL__to_flush = tuple(to_flush)
```

Comment:
What about modifying the constructor of `SQL` to take an iterable instead?
```suggestion
                        to_flush = [f for f in sql.to_flush if f != field]
                        sql = SQL(sql.code, *sql.params, to_flush=to_flush)
```
Alternatively, we could add a method `sql.with_flush(fields)` to do something similar.